### PR TITLE
Calculator-Operations

### PR DESCRIPTION
**Pull request by Alexoid1**
Create a calculate method that handles every action in the calculator.
Create an operation method that handles arithmetic operations.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import operate from './operate';
+
+function calculate(data, buttonName) {
```
Comment (bellom) [inline src/logic/calculate.js:3]: @Alexoid1, Good job so far.

According to the instructions on the pathwright, we need to pass an object as an argument.

The argument passed to the `function calculate` should be an object, we can achieve this by using destructuring.
```
function calculate(data, buttonName) {
  let { total, next, operation } = data;
.....
}
```

image

https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376018/

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import Big from 'big.js';
+
+const operate = (number1, number2, operation) => {
+  const num1 = Big(number1);
+  const num2 = Big(number2);
+  let total = 0;
+
+  const divide = () => (num2 === 0 ? 'NaN' : num1.div(num2));
+
+  const modulus = (num1, num2) => (!num2 ? num1.times(0.01) : num1.times(num2).times(0.01));
```
Comment (bellom) [inline src/logic/operate.js]: Job well done writing `function modulus` but you don't have to do the calculating yourself, that where the use of `Big` comes in -> [big js library](http://mikemcl.github.io/big.js/#mod)

example: case modifier will be -> `num1.mod.(num2)`

### Heads-Up:
You might want to apply `toString()` at the end of your total value.
example: `total = num1.plus(num2).toString();

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @Alexoid1  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @Alexoid1  👋

**Good job here. You've done well fixing the issues raised in the last review. 👌**

However. let's make some fixes.

- Please refer to the comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      total = num1.minus(num2).toString();
+      break;
+    case '÷':
+      total = divide(num1, num2).toString();
+      break;
+    case '%':
+      total = num1.mod(num2).toString();
+      break;
+    default:
+          // code block
```
Comment (mikenath223) [inline src/logic/operate.js]: - [ ] Please include a default here. It shouldn't be empty.

File: src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      total = num1.times(num2).toString();
+      break;
+    case '-':
+      total = num1.minus(num2).toString();
+      break;
+    case '÷':
+      total = divide(num1, num2).toString();
+      break;
+    case '%':
+      total = num1.mod(num2).toString();
```
Comment (mikenath223) [inline src/logic/operate.js:24]: - [ ] Kindly include a condition that checks for decimal(.) points also.

Comment (Alexoid1) [reply src/logic/operate.js:24]: @mikenath223 in the Project specification don't say nothing to set '.' as a operator because is not an operator

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi @Alexoid1 :hand: 
Great job implementing the requested changes.
Good luck with the next milestone :smile: 

Happy coding :computer: 
**Sergio Torres**

**Pull request merged**